File: content/03/code/dotnet/01--step01-literal-match/Shouty/Shouty.Specs/StepDefinitions/StepDefinitions.cs
using System;
using TechTalk.SpecFlow;
using Xunit;

namespace Shouty.Specs.StepDefinitions
{
    [Binding]
    public class StepDefinitions
    {
        private Person lucy = new Person();
        private Person sean = new Person();
        private string messageFromSean;

        [Given("Lucy is located 15 metres from Sean")]
        public void GivenLucyIsLocatedMetresFromSean()
        {
            throw new NotImplementedException("Matched!");
        }

        [When("Sean shouts {string}")]
        public void WhenSeanShouts(string message)
        {
            sean.Shout(message);
            messageFromSean = message;
        }

        [Then("Lucy should hear Sean's message")]
        public void ThenLucyShouldHearSeansMessage()
        {
            Assert.Contains(messageFromSean, lucy.GetMessagesHeard());
        }
    }
}

File: content/bdd-with-cucumber/02/code/java/10-when-stepdef-added/shouty/src/test/java/shouty/StepDefinitions.java
package shouty;

import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class StepDefinitions {
    Person lucy = new Person();
    Person sean = new Person();

    @Given("Lucy is {int} metres from Sean")
    public void lucy_is_metres_from_Sean(Integer distance) {

        lucy.setLocation(distance);
        sean.setLocation(0);
    }

    @When("Sean shouts {string}")
    public void sean_shouts(String message) {
        sean.shout(message):
     }

    @Then("Lucy should hear Sean's message")
    public void lucy_should_hear_Sean_s_message() {
        // Write code here that turns the phrase above into concrete actions
        throw new io.cucumber.java.PendingException();
    }
}
